src/engines/alsadevicefinder.cpp

```diff
@@ -0,0 +1,110 @@
+/* This file is part of Clementine.
+   Copyright 2017, Jonas Kvinge <jonas@jkvinge.net>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <alsa/asoundlib.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <QList>
+#include <QString>
+#include <QVariant>
+
+#include <core/logging.h>
+
+#include "alsadevicefinder.h"
+#include "devicefinder.h"
+
+AlsaDeviceFinder::AlsaDeviceFinder() : DeviceFinder("alsasink") {}
+
+QList<DeviceFinder::Device> AlsaDeviceFinder::ListDevices() {
+  QList<Device> ret;
```

Comment:
We don't have to write C89 here, please move the declarations next to their usages :-)